### PR TITLE
fix1037_paloalto_panos_show_running_nat-policy

### LINKED ISSUE #1037: PaloAlto Show running nat-policy - Source field doesnt read a list of source IP
##### ISSUE TYPE

 - Bug Report
 
##### TEMPLATE USING

```
paloalto_panos_show_running_nat-policy.textfsm
```

##### SAMPLE COMMAND OUTPUT

```
"DMZ-PROXY-NAT; index: 1" {
        nat-type ipv4;
        from DMZ-APPS;
        source [ 10.1.1.1 10.1.1.2 ];
        to UNTRUSTED;
        to-interface ethernet1/5 ;
        destination any;
        service 0:any/any/any;
        translate-to "src: 2.2.2.2 (dynamic-ip-and-port) (pool idx: 1)";
        terminal no;
}
```

##### SUMMARY
show running nat-policy template cant able to parse the source IP list. Tempalte is working if we have single IP in Source, but as soon as we have list its not capturing the data.

Its also not capturing the to-interface

##### STEPS TO REPRODUCE

```yaml

        output = ssh_conn.send_command('show running nat-policy', use_textfsm=True, delay_factor=1)
        print(output)
```

##### EXPECTED RESULTS

{'name': '"DMZ-PROXY-NAT; index: 1"', 'nat_type': 'ipv4', 'from': 'DMZ-APPS', 'source': '[10.1.1.1 10.1.1.2', 'to': 'UNTRUSTED', 'to_interface': 'ethernet1/5', 'destination': 'any', 'service': '0:any/any/any', 'translate_to': 'src: 2.2.2.2 (dynamic-ip-and-port) (pool idx: 1)', 'terminal': 'no'}

##### ACTUAL RESULTS

{'name': '"DMZ-PROXY-NAT; index: 1"', 'nat_type': 'ipv4', 'from': 'DMZ-APPS', 'source': '', 'to': 'UNTRUSTED', 'to_interface': '', 'destination': 'any', 'service': '0:any/any/any', 'translate_to': 'src: 2.2.2.2 (dynamic-ip-and-port) (pool idx: 1)', 'terminal': 'no'}

```

```

### PR DESCRIPTION
**Pull request by diepes**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT

ntc_templates/templates/paloalto_panos_show_running_nat-policy.textfsm

##### SUMMARY

SOURCE changed form single IP to list
add -> Error 

Fixes #1037

```
New raw template.
```

### CODE REVIEW

**Review by alihyder-s — COMMENTED** [review]
I have modified the code for DESTINATION as well.  But Still it cant able to process data if there is list in the fields.

Destination : List of Subnets or List of IPs
Source: List of Subnets or List of IPs
From: List
To: List

**Comment by alihyder-s** [comment]
"Rule 4 Nat-ID 4; index: 8" {
        nat-type ipv4;
        from [ Mgmt Data TRUSTED UNTRUSTED ];
        source [ 10.0.0.0/8 172.0.0.0/16];
        to [ UNTRUSTED DMZ ];
        to-interface ethernet1/5;
        destination [ 1.1.1.1 2.2.2.2 ];
        service 0:any/any/any;
        translate-to "src: 3.3.3.3 (dynamic-ip-and-port) (pool idx: 24)";
        terminal no;

**Comment by diepes** [comment]
@alihyder-s ok so you are saying here is another example where the destination is a list.

For new we just capture the lists as a string, to keep it backwards compatible, also seems PaloAlto invented there own list format with no commas.

I will add another test for source and destination lists.

```
"Rule 4 Nat-ID 4; index: 8" {
nat-type ipv4;
from [ Mgmt Data TRUSTED UNTRUSTED ];
source [ 10.0.0.0/8 172.0.0.0/16 ];
to [ UNTRUSTED DMZ ];
to-interface ethernet1/5;
destination [ 1.1.1.1 2.2.2.2 ];
service 0:any/any/any;
translate-to "src: 3.3.3.3 (dynamic-ip-and-port) (pool idx: 24)";
terminal no;
```

**Comment by diepes** [comment]
@alihyder-s please check updated PR - added new example and test.

File: ntc_templates/templates/paloalto_panos_show_running_nat-policy.textfsm [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+  ^\s+from\s+${FROM};$$
+  ^\s+source\s+${SOURCE};$$ 
+  ^\s+to\s+${TO};$$
+  ^.+to-interface\s${TO_INTERFACE}\s?;$$
+  ^\s+destination\s+${DESTINATION};$$
+  ^\s+service\s+${SERVICE};$$
+  ^\s+translate-to\s+"${TRANSLATE_TO}";$$
+  ^\s+terminal\s+${TERMINAL};$$
+  ^}$$ -> Record
+  ^. -> Error
```
Comment (jmcgill298) [inline ntc_templates/templates/paloalto_panos_show_running_nat-policy.textfsm:25]:
```suggestion
  ^. -> Error

```

**Review by jmcgill298 — CHANGES_REQUESTED** [review]

**Pull request merged**